### PR TITLE
NFDIV-1608/NFDIV-2478

### PR DESCRIPTION
**Pull request by BWeeks-JusticeGovUK**
### Change description ###

Joint hub page for awaiting final order and final order requested 

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1608
https://tools.hmcts.net/jira/browse/NFDIV-2478

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/steps/applicant1/hub-page/right-column/content.ts [hunk 1]
```diff
@@ -142,27 +147,31 @@ const languages = {
...
+  const hasCertificateOfEntitlement = userCase.coCertificateOfEntitlementDocument;
+  const hasConditionalOrderGranted = userCase.coConditionalOrderGrantedDocument;
+  const hasConditionalOrderAnswers = userCase.documentsGenerated?.find(
     doc => doc.value.documentType === DocumentType.CONDITIONAL_ORDER_ANSWERS
   );
-  const hasConditionalOrderApplication = content.userCase.documentsGenerated?.find(
+  const hasConditionalOrderApplication = userCase.documentsGenerated?.find(
     doc => doc.value.documentType === DocumentType.CONDITIONAL_ORDER_APPLICATION
   );
+  const hasFinalOrderApplication =
```
Comment (danielbe11) [inline src/main/steps/applicant1/hub-page/right-column/content.ts:172]: Feel like this name might be a bit misleading - the condition is checking that we both have FINAL_ORDER_APPLICATION document and we are in FinalOrderRequested state.

File: src/test/functional/features/joint-final-order.feature [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    When I enter my valid case reference and valid access code
+    Then the page should include "You need to review your joint application"
+    Given I've already completed the form using the fixture "jointApplicant2CompleteCase" for "applicant2"
+    And I go to "/applicant2/confirm-your-joint-application"
+    Given I select "I confirm that I’m applying to the court to dissolve my marriage (get a divorce)"
+    And I select "I believe that the facts stated in this application are true"
+    When I click "Submit"
+    Then the page URL should be "/applicant2/needs-to-confirm-joint-application"
+
+#  //TODO when final order is implemented change test to user path instead of changing the states for each page
```
Comment (danielbe11) [inline src/test/functional/features/joint-final-order.feature:18]: Might be best to make a ticket for this and add the number here so we don't forget about this.

File: src/main/steps/applicant1/hub-page/right-column/content.ts (hunk 1 above)
Comment (BWeeks-JusticeGovUK) [reply src/main/steps/applicant1/hub-page/right-column/content.ts]: Altered to hasFinalOrderApplicationAndFinalOrderRequested

File: src/test/functional/features/joint-final-order.feature (hunk 2 above)
Comment (BWeeks-JusticeGovUK) [reply src/test/functional/features/joint-final-order.feature]: Refactored the tests into the hub page feature tests, so this TODO should not be necessary.

**Review by danielbe11 — APPROVED** [review]

File: src/main/steps/applicant1/hub-page/joint/content.ts [hunk 3]
```diff
@@ -215,6 +247,9 @@ export const generateContent: TranslationFn = content => {
   const displayState = currentStateFn(userCase).at(
     (userCase.state === State.OfflineDocumentReceived ? userCase.previousState : userCase.state) as State
   );
+  const hasApplicantAppliedForFinalOrder = isApplicant2
+    ? userCase.applicant2AppliedForFinalOrderFirst === YesOrNo.YES
+    : userCase.applicant1AppliedForFinalOrderFirst === YesOrNo.YES;
```
Comment (lara-mpt) [inline src/main/steps/applicant1/hub-page/joint/content.ts:252]: think a better name for the variable would be hasApplicantAppliedForFinalOrderFirst

File: src/main/steps/applicant1/hub-page/joint/jointTemplateSelector.test.ts [hunk 4]
```diff
@@ -59,9 +59,15 @@ describe('JointTemplateSelector test', () => {
     expect(jointTemplate).toBe('/applicant-not-yet-applied-for-conditional-order.njk');
   });
 
-  test('should show /awaiting-final-order.njk for state AwaitingFinalOrder', () => {
+  test('should show /awaiting-final-order.njk for states in AwaitingFinalOrder', () => {
```
Comment (lara-mpt) [inline src/main/steps/applicant1/hub-page/joint/jointTemplateSelector.test.ts:62]: not sure this sentence makes sense?

**Review by lara-mpt — APPROVED** [review]
approved, but left a couple small comments

File: src/main/steps/applicant1/hub-page/joint/content.ts (hunk 3 above)
Comment (velimd) [reply src/main/steps/applicant1/hub-page/joint/content.ts:252]: yh, tbh im only using that field as it seems its the only way to determine if they have applied for final order in joint.

**Pull request merged**